src/Microsoft.SymbolStore.Client/CabStream.cs

```diff
@@ -0,0 +1,265 @@
+        private Stream _stream;
+        private FileOpenDelegate _open;
+        private FileCloseDelegate _close;
+        private FileSeekDelegate _seek;
+        private FileReadDelegate _read;
+        private FileWriteDelegate _write;
+        private MemAllocDelegate _alloc;
+        private MemFreeDelegate _free;
+        private Error _error = new Error();
+        private IntPtr _outputHandle;
+
+        public CabConverter(Stream stream)
+        {
+            _stream = stream;
+            
+            _open = Open;
+            _close = Close;
+            _seek = Seek;
+            _read = Read;
+            _write = Write;
+            _alloc = Alloc;
+            _free = Free;
+
+        }
+
+        public MemoryStream Convert()
+        {
+            if (_stream is MemoryStream)
+            {
+                _input = (MemoryStream)_stream;
+            }
+            else
+            {
+                _input = new MemoryStream();
+                _stream.CopyTo(_input);
+            }
+
+            _input.Position = 0;
+
+            Begin();
+            _output.Position = 0;
+            return _output;
+        }
+
+        public async Task<MemoryStream> ConvertAsync()
+        {
+            if (_stream is MemoryStream)
+            {
+                _input = (MemoryStream)_stream;
+            }
+            else
+            {
+                _input = new MemoryStream();
+                await _stream.CopyToAsync(_input);
+            }
+
+            _input.Position = 0;
+
+            Task task = new Task(() => Begin());
+            task.Start();
+
+            await task;
+            _output.Position = 0;
+            return _output;
+        }
+
+        public void Begin()
+        {
+            _context = Create(_alloc, _free, _open, _read, _write, _close, _seek, -1, _error);
+            Copy(_context, "", "", 0, Notify, IntPtr.Zero, IntPtr.Zero);
+            Destroy(_context);
+        }
+
+        private IntPtr CreateStreamPointer(MemoryStream ms)
+        {
+            return (IntPtr)GCHandle.Alloc(ms);
+        }
+        
+        private MemoryStream PointerToStream(IntPtr ptr)
+        {
+            GCHandle handle = (GCHandle)ptr;
+            return (MemoryStream)handle.Target;
+        }
+
+        private void FreeStreamPointer(IntPtr ptr)
+        {
+            GCHandle handle = (GCHandle)ptr;
+            handle.Free();
+        }
+
+        string ToHex(IntPtr ptr)
+        {
+            return ptr.ToString("x");
+        }
+
+        private IntPtr Notify(NotificationType type, Notification notification)
+        {
+            if (type != NotificationType.CopyFile)
+                return IntPtr.Zero;
+
+            switch (type)
+            {
+                case NotificationType.CopyFile:
+                    _outputHandle = CreateStreamPointer(_output);
+                    Debug.WriteLine($"Notify({type}, {notification}) - {ToHex(_outputHandle)}");
```

Comment:
Can we take these Debug.WriteLines out or put them behind a conditional? I worry that people debugging an app that happens to use this class will need to deal with this spew but it won't relate to the thing they are trying to debug.